cpp/build-support/convert-benchmark.py

```diff
@@ -0,0 +1,299 @@
+        # arrow features
+        deps = {}
+        toolchain = {
+            "language_implementation_version": "c++11",
+            "dependencies": deps,
+        }
+        return toolchain
+
+    def as_arrow(self):
+        ctx = {
+            "benchmark_language": "C++",
+            "run_timestamp": self.date,
+        }
+
+        for extra in (self.host, self.git, self.toolchain):
+            ctx.update(extra)
+
+        return ctx
+
+    @classmethod
+    def from_json(cls, version, payload):
+        return cls(**payload)
+
+
+class BenchmarkObservation:
+    def __init__(self, version, **kwargs):
+        self._name = kwargs.get("name")
+        self.version = version
+        self.real_time = kwargs.get("real_time")
+        self.cpu_time = kwargs.get("cpu_time")
+        self.time_unit = kwargs.get("time_unit")
+        self.size = kwargs.get("size")
+        self.bytes_per_second = kwargs.get("bytes_per_second")
+
+    @property
+    def is_mean(self):
+        return self._name.endswith("_mean")
+
+    @property
+    def is_median(self):
+        return self._name.endswith("_median")
+
+    @property
+    def is_stddev(self):
+        return self._name.endswith("_stddev")
+
+    @property
+    def is_agg(self):
+        return self.is_mean or self.is_median or self.is_stddev
+
+    @property
+    def is_realtime(self):
+        return self.name.find("/realtime") != -1
+
+    @property
+    @memoize
+    def name(self):
+        name = self._name
+        return name.rsplit("_", maxsplit=1)[0] if self.is_agg else name
+
+    @property
+    def value(self):
+        """ Return the benchmark value."""
+        if self.size:
+            return self.bytes_per_second
+        return self.real_time if self.is_realtime else self.cpu_time
+
+    @property
+    def unit(self):
+        if self.size:
+            return "bytes_per_second"
+        return self.time_unit
+
+    def __str__(self):
+        return f"BenchmarkObservation[name={self.name}]"
+
+
+class BenchmarkSuite:
+    def __init__(self, name, version, runs):
+        self.name = name
+        self.version = version
+        # exclude google benchmark aggregate artifacts
+        aggs, runs = partition(lambda b: b.is_agg, runs)
+        self.runs = sorted(runs, key=lambda b: b.value)
+        self.values = [b.value for b in self.runs]
+        self.aggregates = aggs
+
+    @property
+    def n_runs(self):
+        return len(self.runs)
+
+    @property
+    @memoize
+    def mean(self):
+        maybe_mean = [b for b in self.aggregates if b.is_mean]
+        if maybe_mean:
+            return maybe_mean[0].value
+        # fallback
+        return sum(self.values) / self.n_runs
+
+    @property
+    @memoize
+    def median(self):
+        maybe_median = [b for b in self.aggregates if b.is_median]
+        if maybe_median:
+            return maybe_median[0].value
+        # fallback
+        return self.runs[int(self.n_runs / 2)].value
```

Comment:
That's correct only if `n_runs` is odd (otherwise it should probably be the average of the two "middle" values)... Is there a circumstance where gbenchmark does several runs but doesn't output the median / mean / stddev ?

Reply:
I added the statistics just for the sake of completeness with min/q1/q3/max (which are not provided by google benchmark). According to wikipedia, the median is `n*n+1/2` when odd, I'll fix it.